### ISSUE TITLE
Client map download not working properly

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#823]
Author: @IR4T4 (IR4T4)
Date: 2015-02-23
Redmine Issue: 823, https://dev.etlegacy.com/issues/823

---

I've noticed a bug in client-server download process. Sometimes the client doesn't connect after map downloads while the server shows client already as connected. 

Test case:
- open a listen server and put some custom maps into fs_basepath (more than 5 so other client is forced to download) 
- connect with another client (fresh and clean fs_homepath), client should start a download 
- start another map f.e. oasis
- after warmup when SV_MapRestart_f is called the other client goes from connecting to connected *boom*

/reconnect solves the issue but isn't a solution ...

*Line 383 in SV_MapRestart_f of sv_ccmds.c calls SV_ClientEnterWorld(client, NULL); which sets CS_ACTIVE and causes our issue.*

---

Related issue(s): #822

---

Related commit(s):
f56e13bd0a4dd6a0c292085ff0d503ce9f4ae86d - general: Block client from entering the game if do..

**Comment by ETLegacyTeam** [comment]
From: Zecklar (Zecklar)
Date: 2015-03-19T18:52:56Z

---

I tested it and there was no issues anymore on my side. Also now the player status stays as "Connecting" after the map start if the player is still downloading. I'd like someone to confirm that there is no *boom* anymore.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2015-03-19T18:55:30Z

---

Alright, let's also check vanilla clients because the change affects them too.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2015-04-16T19:02:43Z

---

I can't reproduce this anymore. 

{TODO}
- run test case with vanilla client and etl server

**Comment by ETLegacyTeam** [comment]
From: @Zelly (Zelly)
Date: 2015-05-11T21:44:54Z

---

I can reproduce getting stuck loading map with latest build.
There were no downloads involved.
I can tell it is attempting to retry to load because it flashes the console very slightly every second or so.
I used rcon to change map right before I clicked join on my client, but it has happened before with a normal rotation switch.
Tested on silent mod etl client, et300 server.
